grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -96,7 +95,7 @@ class ProductController {
         [products: product]
     }
 
-
+    @Transactional
```

Comment:
Is this batchSave method being used somewhere?

Reply:
I just realized that you're just applying @Transactional to all methods so you can ignore this comment.